metaserver/src/main/scala/scala/meta/languageserver/search/IndexDependencyClasspath.scala

```diff
@@ -0,0 +1,49 @@
+package scala.meta.languageserver.search
+import java.util.concurrent.ConcurrentHashMap
+import scala.meta.languageserver.ScalametaLanguageServer.cacheDirectory
+import scala.meta.languageserver.compiler.CompilerConfig
+import scala.meta.languageserver.{Effects, ServerConfig, ctags}
+import scala.meta.languageserver.storage.LevelDBMap
+import com.typesafe.scalalogging.Logger
+import monix.reactive.Observer
+import org.langmeta.internal.semanticdb.schema.{Database, Document}
+import org.langmeta.io.AbsolutePath
+
+object IndexDependencyClasspath {
+  // NOTE(olafur) this probably belongs somewhere else than Compiler, see
+  // https://github.com/scalameta/language-server/issues/48
+  def apply(
+      serverConfig: ServerConfig,
+      indexedJars: ConcurrentHashMap[AbsolutePath, Unit],
+      documentSubscriber: Observer.Sync[Document],
+      logger: Logger,
+      sourceJars: List[AbsolutePath]
+  ): Effects.IndexSourcesClasspath = {
+    if (!serverConfig.indexClasspath) return Effects.IndexSourcesClasspath
```

Comment:
Can we avoid passing in serverConfig by delegating this to the call site?

Reply:
Do you mean check if serverConfig.indexClasspath is false, and if so just return Effects.IndexSourcesClasspath from ScalacProvider directly instead of calling IndexDependencyClasspath?